src/hitchhiker/tree/core.clj

```diff
@@ -102,11 +174,13 @@
   [children]
   (mapv last-key (butlast children)))
 
+(declare ->IndexNode)
+
 (defrecord IndexNode [children storage-addr op-buf cfg]
   IResolve
   (index? [this] true)
-  (dirty? [this] (not (realized? storage-addr)))
-  (resolve [this] this) ;;TODO this is a hack for testing
+  (dirty? [this] (not (async/poll! storage-addr)))
+  (resolve [this] (go this)) ;;TODO this is a hack for testing
```

Comment:
I don't think it's a hack any more, then :) 

In the past, this may have been more complex than just returning this. Would you remove that comment?